File: cambiarFechas.py
import os
import json
import sys
from datetime import datetime

def fix_datetime(path=os.getcwd()):
    extensions = ['gif', 'png', 'jpg', 'jpeg', 'tiff', 'mp4', 'avi', 'mov', 'flv']

    files = []
    for root, dirs, filenames in os.walk(path):
        for filename in filenames:
            if any(filename.lower().endswith(ext) for ext in extensions):
                files.append(os.path.join(root, filename))

    for file in files:
        file_path, file_name = os.path.split(file)
        file_name_without_ext, extension = os.path.splitext(file_name)

        if file_name_without_ext.endswith(')'):
            index = file_name_without_ext[-3:]
            if index == '(0)':
                index = ''
        else:
            index = ''

        json_file = file_name_without_ext.replace(index, '').replace('-editado', '')
        json_file_path = os.path.join(file_path, f"{json_file}.{extension}{index}"[:46] + '.json')

        if os.path.exists(json_file_path):
            with open(json_file_path, 'r') as f:
                json_data = json.load(f)

            if 'photoTakenTime' in json_data and 'timestamp' in json_data['photoTakenTime']:
                new_time = int(json_data['photoTakenTime']['timestamp'])
                try:
                    os.utime(file, (new_time, new_time))
                except OSError:
                    print(f"Failed to update modified time for {file}.")
        else:
            print(f'File not found: {json_file_path}')
            continue

if __name__ == "__main__":
    if len(sys.argv) == 1:
        fix_datetime()
    else:
        fix_datetime(sys.argv[1])

File: cambiarFechas.php
<?php

if (count($argv) == 1) {
	fixDatetime();
} 
else {
	fixDatetime($argv[1]);
}

function fixDatetime($path = __DIR__) {
	// Usar RecursiveIterator
	$extensions = array('gif', 'png', 'jpg', 'jpeg', 'tiff', 'mp4', 'avi', 'mov', 'flv');

	$files = array();
	$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($path));
	foreach ($iterator as $file) {
		if ($file->isFile() && in_array(pathinfo($file, PATHINFO_EXTENSION), $extensions)) {
			$files[] = $file->getPathname();
		}
	}

	foreach ($files as $file) {
		$filePath = pathinfo($file, PATHINFO_DIRNAME);
		$fileName = pathinfo($file, PATHINFO_FILENAME);
		$extension = pathinfo($file, PATHINFO_EXTENSION);

		if (substr($fileName, -1) == ')') {
			$indice = substr($fileName, -3);

			if ($indice == '(0)') {
				$indice = '';
			}
		}
		else {
			$indice = '';
		}

		$jsonFile = str_replace($indice, '', $fileName);
		$jsonFile = str_replace('-editado', '', $jsonFile);
		
		$jsonFilePath = $filePath . DIRECTORY_SEPARATOR . substr($jsonFile . '.' . $extension . $indice, 0, 46) . '.json';

		if (file_exists($jsonFilePath)) {
			$jsonData = file_get_contents($jsonFilePath);
			$jsonData = json_decode($jsonData, true);

			if (isset($jsonData['photoTakenTime']['timestamp'])) {
				$new_time = date('U', $jsonData['photoTakenTime']['timestamp']);

				if (!touch($file, $new_time)) {
					echo "Failed to update modified time to {$file}.";
				}
			}
		}
		else {
			echo 'File not found: ' . $jsonFilePath . "\n";
			continue;
		}
	}
}